tests/integration/test_templates.py

```diff
@@ -93,34 +95,41 @@ async def test_receiver_gets_alert(ops_test: OpsTest, charm_under_test, httpserv
         }
     ]
 
-    # define the expected slack notification for the alert
-    expected_notification = {
-        "channel": "test",
-        "username": "Alertmanager",
-        "attachments": [
-            {
-                "title": f"[FIRING:1] {alert_name} {app_name} {ops_test.model_name} {model_uuid} ",
-                "title_link": f"http://{app_name}-0:9093/#/alerts?receiver={receiver_name}",
-                "text": f"https://localhost/alerts/{alert_name}",
-                "fallback": f"[FIRING:1] {alert_name} {app_name} {ops_test.model_name} {model_uuid}  | "
-                f"http://{app_name}-0:9093/#/alerts?receiver={receiver_name}",
-                "callback_id": "2",
-                "footer": "",
-                "color": "danger",
-                "mrkdwn_in": ["fallback", "pretext", "text"],
-            }
-        ],
-    }
+    request_from_alertmanager = None
+
+    def request_handler(request: Request):
+        """A request handler.
+
+        Alertmanager's POST request to a slack server looks like this:
+
+        {'attachments': [{'callback_id': '2',
+                  'color': 'danger',
+                  'fallback': '[FIRING:1] fake-alert alertmanager-k8s '
+                              'test-templates-klzm 1234  | '
+                              'http://alertmanager-k8s-0.fqdn:9093/#/alerts?receiver=name',
+                  'footer': '',
+                  'mrkdwn_in': ['fallback', 'pretext', 'text'],
+                  'text': 'https://localhost/alerts/fake-alert',
+                  'title': '[FIRING:1] fake-alert alertmanager-k8s '
+                           'test-templates-klzm 1234 ',
+                  'title_link': 'http://alertmanager-k8s-0.fqdn:9093/#/alerts?receiver=name'}],
+        'channel': 'test',
+        'username': 'Alertmanager'}
+        """
+        nonlocal request_from_alertmanager
+        response = Response("OK", status=200, content_type="text/plain")
+        request_from_alertmanager = json.loads(request.data.decode("utf-8"))
+        logger.info("Got Request Data : %s", request_from_alertmanager)
+        return response
 
     # set the alert
     with httpserver.wait(timeout=120) as waiting:
         # expect an alert to be forwarded to the receiver
-        httpserver.expect_oneshot_request(
-            "/", method="POST", json=expected_notification
-        ).respond_with_handler(request_handler)
+        httpserver.expect_oneshot_request("/", method="POST").respond_with_handler(request_handler)
         client_address = await get_unit_address(ops_test, app_name, 0)
         amanager = Alertmanager(address=client_address)
         amanager.set_alerts(alerts)
 
     # check receiver got an alert
     assert waiting.result
+    assert request_from_alertmanager["attachments"][0]["callback_id"] == callback_id  # type: ignore
```

Comment:
I presume you have checked that this test can be made to fail by changing the expected `callback_id` to be different from the `callback_id` that is set by the template.